pipeline/ingest/behavior_ingest.py

```diff
@@ -0,0 +1,89 @@
+import datajoint as dj
+
+from pipeline import lab, experiment
+from pipeline import get_schema_name
+from . import session_loaders
+
+schema = dj.schema(get_schema_name('behavior_ingest'))
+
+# ============== SETUP the LOADER ==================
+
+try:
+    data_dir = dj.config['custom']['data_root_dir']
+except KeyError:
+    raise KeyError('Unspecified data root directory! Please specify "data_root_dir" under dj.config["custom"]')
+
+try:
+    session_loader_class = dj.config['custom']['session_loader_class']
+except KeyError:
+    raise KeyError('Unspecified session loader method! Please specify "session_loader_method" under dj.config["custom"]')
+
+if session_loader_class in session_loaders.__dict__:
+    # instantiate a loader class with "data_dir" and "config" (optional)
+    loader_class = session_loaders.__dict__[session_loader_class]
+else:
+    raise RuntimeError(f'Unknown session loading function: {session_loader_class}')
+
+# instantiate a loader class with "data_dir" and "config" (optional)
+session_loader = loader_class(data_dir, dj.config)
+
+# ============== BEHAVIOR INGESTION ==================
+
+
+@schema
+class InsertedSession(dj.Imported):
+    definition = """  # Ingestion-specific information about a session
+    -> experiment.Session
+    ---
+    loader_method: varchar(16)  # name of loader method used
+    sess_data_dir: varchar(255) # directory path for this session, relative to root data directory
+    """
+
+    class SessionFile(dj.Part):
+        definition = """  # file(s) associated with a session
+        -> master
+        filepath: varchar(255)  # relative filepath with respect to root data directory
+        """
+
+
+def load_all_sessions(subject_id):
+    # ---- parse data dir and load all sessions ----
+    try:
+        sessions_to_ingest = session_loader.load_sessions(data_dir, subject_id)
```

Comment:
Why is `data_dir` necessary again if already used in the instantiation above in `loader_class`?

Reply:
thanks for the catch, no need for `data_dir` here with the change in implementation from function to class